### ISSUE TITLE
eq-without-hash check removed

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#5025]
### Bug description

```python
class A:
    def __init__(self) -> None:
        self.x = 5

    def __eq__(self, other: object) -> bool:
        return isinstance(other, A) and other.x == self.x
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
empty
```

### Expected behavior

Formerly `eq-without-hash` error was issued, but not more since pylint 2.11.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04.3 LTS \n \l

**Comment by cdce8p** [comment]
The warning was removed as part of #4942.
@Pierre-Sassoulas What do you think? Maybe this should be added back as an optional checker?

**Comment by jolaf** [comment]
I saw #4942, but in Python 3, automatic reset of '__hash__` to `None` if `__eq__` is defined is non-trivial behavior that may cause hard-to-catch bugs.
I think its pretty reasonable to have a capability to warn about such things.
Even more so as this check was already present.
In other words, from my point of view, this check is usable even in strictly Python 3 context.

I didn't check, but I suspect some other checks that were introduced for checking the correctness of porting, and are now removed, are in fact pretty useful in Python 3-only code too.

**Comment by cdce8p** [comment]
@jolaf I went looked through the list twice, once for the review and the second time for your issue. `eq-without-hash` is the only one I know about now, all others seem to check for Python 2 behavior that isn't possible in Python 3. Tbh I haven't used Python 2, so it might be possible that I missed others. You can find the list here:
https://github.com/PyCQA/pylint/pull/4942/files#diff-34a57926d10bd84d60b6ddceb5447be233469d5e4e097ce7e5ecd64287fc5899L188-L608

Keep in mind that all of them had been disabled by default already. Your issue is the first one we've gotten about this.

**Comment by jolaf** [comment]
@cdce8p Yeah, I think you're right.
I've now also checked the list (thanks for the clear link) and found no other cases raising suspicions.

**Comment by Pierre-Sassoulas** [comment]
Let's move that to an existing checker, I think it could even be a default check ? Thank you for pointing that out @jolaf I did not use it myself as everything in the python3 porting checker was disabled by default.

**Comment by Pierre-Sassoulas** [comment]
It was restored in #5843 following this very issue that argue for it to be reinstated.

**Comment by Pierre-Sassoulas** [comment]
You need to activate the corresponding extension now, see notes from the link you pasted:
> This message is emitted by the optional ['eq-without-hash'](https://pylint.pycqa.org/en/latest/user_guide/checkers/extensions.html#pylint-extensions-eq-without-hash) checker which requires the ``pylint.extensions.eq_without_hash`` plugin to be loaded.
